networkit/cpp/io/KONECTGraphReader.cpp

```diff
@@ -1,226 +1,279 @@
+				directed = false;
+			}else if(graphFormat != "asym"){
+				throw std::runtime_error("Graph format not tagged properly. Format is: " + graphFormat);
+			}
+			skipWhitespace();
+			// Parse graph weighting
+			graphType = scanWord();
+			//NOTE: This parser disregards the edge weight classification of "interval scale" and "ratio scale"
+			if (graphType == "weighted" || graphType == "posweighted" || graphType == "signed"){
+				weighted = true;
+				valuesPerLine = 3;
+			} else if (graphType == "positive"){ //multiple edges
+				if(multipleEdgesHandlingMethod == SUM_WEIGHTS_UP){
+					weighted = true;
+				}
+				multiple = true; //weights will be added
+			} else if (graphType == "multisigned" || graphType == "multiweighted" || graphType == "multiposweighted"){
+				weighted = true;
+				multiple = true;
+				valuesPerLine = 3;
+			} else if (graphType == "dynamic"){
+				throw std::runtime_error("Dynamic graphs are not supported yet");
+			} else if (graphType != "unweighted"){
+				throw std::runtime_error("Graph weight not tagged properly. Weight is: " + graphType);
+			}
+			DEBUG("First property line read in. Format: "+graphFormat+" / Type: "+graphType);
+			if (graphFormat == "bip"){
+				INFO("Your graph is flagged as a bipartite one. Keep in mind that"
+				 "NetworKit cannot handle this kind of graph in its special cases."
+				 "It is imported as a usual undirected graph and edges might be discarded.");
+			}
+			if(multiple){
+				DEBUG("Selected handling method for multiple edges is: "+std::to_string(multipleEdgesHandlingMethod));
+			}
+		} else {
+			throw std::runtime_error("No graph properties line found. This line is mandatory.");
+		}
+
+		skipWhitespace();
+		if (*it != '\n'){
+			throw std::runtime_error("No break symbol after first property line");
+		}else{
+			++it;
+		}
+		skipWhitespace();
+		//second optional property line
+		if(*it == '%'){
+			++it;
+			skipWhitespace();
+			numberOfEdges = scanId();
+			skipWhitespace();
+			numberOfNodes = scanId();
+			secondPropertyLine = true;
+			while(it != end && *it != '\n')
+				++it;
+			if(it >= end){ // proper error message if file ends unexpected
+				throw std::runtime_error("Unexpected end of file");
+			}
+			if (*it != '\n'){
+				throw std::runtime_error("File not properly formatted. Last character in second property line is: "+std::string(1, *it));
+			}else{
+				++it;
+			}
+			DEBUG("Second property line read in. Edges: "+std::to_string(numberOfEdges)+ " / Nodes: "+std::to_string(numberOfNodes));
+		}
+
+
+		Graph graph((secondPropertyLine ? numberOfNodes : 0), weighted, directed);
+
+		//Map nodes and increase graph size if no second property is defined
+		std::function<node(node)> mapNode = [&] (node in) -> node {
+			if(secondPropertyLine){
+				if (in > numberOfNodes){ //if file is corrupted
+					//secondPropertyLine is made useless
+					ERROR("Given number of nodes by file does not match actual graph");
```

Comment:
Wouldn't it be better to just throw an exception here? Or is it so common that this property line is wrong? Users might not notice this message and then get wrong results as their graph was broken.

Reply:
It is indeed common. IMHO ERROR is reasonably easy to notice; I don't see what we could do better here.

Reply:
In addition, this behaviour is consistent with the one of the old KONECTGraphReader so we might stick with it.